File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.swift
import Foundation

func solution(_ a:Int, _ b:Int) -> Int {
    // 최대 공약수
    let gcd = (a > b) ? getGCD(a, b) : getGCD(b, a)
    // 분모
    var y = b / gcd
    // 유한 소수 판별
    
    while y != 1 {
        for num in 2...y {
            if y.isMultiple(of: num) {
                if num == 2 || num == 5 {
                    y /= num
                    break
                } else {
                    return 2
                }
            }
        }
    }
    
    return 1
}

func getGCD(_ x: Int, _ y: Int) -> Int {
    // x > y
    var a = x
    var b = y
    var r = a % b
    
    while r != 0 {
        a = b
        b = r
        r = a % b
    }

    return b
}

File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.cpp
#include <string>
#include <vector>

using namespace std;

int GCD(int a, int b) {
    int c = a % b;

    while (c != 0) {
        a = b;
        b = c;
        c = a % b;
    }

    return b;
}

int solution(int a, int b) {
    int answer = 1;


    int _gcd = a > b ? GCD(a, b) : GCD(b, a);

    b /= _gcd;

    if (b == 1) {
        return 1;
    }

    while(b % 2 == 0) { b /= 2; }
    while(b % 5 == 0) { b /= 5; }

    return b != 1 ? 2 : 1;

}